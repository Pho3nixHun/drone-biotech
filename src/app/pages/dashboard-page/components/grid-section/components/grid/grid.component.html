<div class="grid gap-4 grid-cols-subgrid col-span-full">
    <ng-content />
</div>

<!-- It projects <div>-s and <app-grid-item>-s. -->
