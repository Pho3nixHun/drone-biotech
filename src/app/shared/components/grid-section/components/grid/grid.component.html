<div class="grid gap-4 col-span-full">
    <ng-content select="[app-grid-header, app-grid-item-list]" />
</div>
