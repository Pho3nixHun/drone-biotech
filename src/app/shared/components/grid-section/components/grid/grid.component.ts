import { Component } from '@angular/core';
@Component({
    selector: 'app-grid',
    imports: [],
    templateUrl: './grid.component.html',
})
export class GridComponent {}
