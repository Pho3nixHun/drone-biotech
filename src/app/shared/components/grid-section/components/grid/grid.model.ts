export interface GridVM {
    gridColsLength: GridColsLength;
}

export enum GridColsLength {
    SIX = 'grid-cols-6',
    SEVEN = 'grid-cols-7',
}
