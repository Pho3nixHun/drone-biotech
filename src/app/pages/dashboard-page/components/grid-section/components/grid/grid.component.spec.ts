import { ComponentFixture, TestBed } from '@angular/core/testing';
import { GridComponent } from './grid.component';
import { Component } from '@angular/core';

@Component({
    imports: [GridComponent],
    template: `
        <app-grid>
            <p>Should be projected</p>
        </app-grid>
    `,
})
class TestHostComponent {}

describe('GridComponent', () => {
    let fixture: ComponentFixture<TestHostComponent>;
    let compiled: HTMLElement;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [TestHostComponent],
        }).compileComponents();

        fixture = TestBed.createComponent(TestHostComponent);
        compiled = fixture.debugElement.nativeElement;
    });

    // Snapshot testing
    it('should render the template correctly', () => {
        // There is no need to arrange

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });
});
