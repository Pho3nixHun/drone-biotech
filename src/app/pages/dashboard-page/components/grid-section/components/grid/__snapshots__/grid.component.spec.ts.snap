// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridComponent should render the template correctly 1`] = `
<div
  id="root0"
  ng-version="19.2.9"
>
  <app-grid>
    <div
      class="grid gap-4 grid-cols-subgrid col-span-full"
    >
      <p>
        Should be projected
      </p>
    </div>
  </app-grid>
</div>
`;
